### ISSUE TITLE
PubSub items still end up in wrong node sometimes

### ISSUE THREAD

**Issue by mrfrostfire** [processone/ejabberd#1563]
> What version of ejabberd are you using?

16.12

> What operating system (version) are you using?

RedHat

> How did you install ejabberd (source, package, distribution)?

rpm

> What did not work as expected? Are there error messages in the log? What
> was the unexpected behavior? What was the expected result?

Same issue that was fixed in issue #1320 is still occuring.
We upgraded to 16.12 where the fix was applied in january and started from a clean database.
Today we have a pubsub item again in the mnesia db that was created feb 16th and does not belong to any node. Its nodeid is 30, but the highest nodeid we have in the db currently is 22.
If we add more nodes, the node who gets id 30 will get the item read back to its subscribers (i've tested this).
The orphaned items table is empty.

In other words, the issue still persists.
This is causing us a lot of issues again and is a blocker for any new deploy, as our code has been rewritten to use pubsubitem updates (removal of old item > publish new item).

**Comment by mrfrostfire** [comment]
`{pubsub_item,{<<"5CF59DD4E263">>,30},
             {{1487,23005,320715},
              {<<"abcdefghij">>,<<"abcdefg.companyabc.com">>,<<>>}},
             {{1487,23005,320715},
              {<<"abcdefghij">>,<<"abcdefg.companyabc.com">>,
               <<"ABCDEFG123">>}},
             [{xmlel,<<"Communication.Models.EndUser">>,
                     [{<<"xmlns">>,<<>>}],
                     [{xmlel,<<"CustomerName">>,[],
                             [{xmlcdata,<<"Abcdefg Abcdefghi">>}]},
                      {xmlel,<<"Partner">>,[],[{xmlcdata,<<"ABCD">>}]},
                      {xmlel,<<"Email">>,[],
                             [{xmlcdata,<<"abcdefghij@companyabc.com">>}]},
                      {xmlel,<<"Phone">>,[],[{xmlcdata,<<"1234567890">>}]},
                      {xmlel,<<"UniqueID">>,[],
                             [{xmlcdata,<<"12345678">>}]}]}]}.
`

That's the rogue item in the database, if that helps.
I have removed any real data with abc/123..

**Comment by cromain** [comment]
OK, there is a possible leak of orphan items when  removing node affiliation/subscription. I'll fix this now. 
Other point, are you running in cluster ?

**Comment by cromain** [comment]
If you're in cluster, you must make sure pubsub_orphan table is replicated of course. Then i don't see other possible leak after a quick review.

**Comment by mrfrostfire** [comment]
We are running most environments in a cluster. This specific environment though is not running in a cluster.

**Comment by zeropopular** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  mrniceguyzero@gmail.com
    Domain thezeroworld.com has exceeded the max emails per hour (7/7 (100%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext4.iad.github.net ([192.30.252.195]:38834 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by node87.certushost.com with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.88)
	(envelope-from <noreply@github.com>)
	id 1ch27u-00035T-VO
	for zero@thezeroworld.com; Thu, 23 Feb 2017 17:44:43 -0500
Date: Thu, 23 Feb 2017 07:36:30 -0800
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1487864190;
	bh=F5yPBz4lawEmQzxTl0gmoDMuBQEoqTM4YEXkQI7wK5A=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=an2L0/iqId1Ow7u3VixsjO2T9cF8GNLTeqSuFywgGmbNLn7u0b1za8sRDvr4yRBrD
	 n6PghrWo4tmAPTzmUxNBcC9bteRQian9RaOB0sEWz2pJGHloj918XVxIwikJOBKPlm
	 SSWS4qBwF2whs4IN9342RcrSnMNEHf40TxmtAsEw=
